tests/testthat/test-SCD-effect-sizes.R

```diff
@@ -193,19 +197,127 @@ test_that("Batch calculator is correct", {
                     std_dev = "baseline",
                     confidence = 0.95,
                     pretest_trend = FALSE,
-                    format = "long")
+                    format = "long",
+                    warn = FALSE
     ) %>%
     mutate(across(Est:CI_upper, ~ round(., 2)))
+  
+  data(Wright2012)
+  Wright_pkg <-
+      batch_calc_ES(dat = Wright2012,
+                    grouping = c(Participant),
+                    condition = Condition,
+                    outcome = Prosocial_behavior,
+                    session_number = Session,
+                    baseline_phase = "baseline",
+                    intervention_phase = "intervention A",
+                    ES = all_names,
+                    improvement = "increase",
+                    pct_change = FALSE,
+                    scale = "count",
+                    std_dev = "baseline",
+                    confidence = 0.95,
+                    pretest_trend = FALSE,
+                    format = "long",
+                    warn = FALSE
+    ) %>%
+    mutate(across(Est:CI_upper, ~ round(., 2))) %>% 
+    mutate(Participant = as.character(Participant))
 
-  expect_equal(output_pkg, output_app_table, check.attributes = FALSE)
+  expect_equal(McKissick_pkg, McKissick_app, check.attributes = FALSE)
+  expect_equal(Schmidt_pkg, Schmidt_app, check.attributes = FALSE)
+  expect_equal(Wright_pkg, Wright_app, check.attributes = FALSE)
 
 })
 
+
+
+
+# Check data uploading
+check_load <- function(file) {
+
+  # myfile <- tempfile()
+  # write.csv(file, file = myfile)
+
+  data_path <- paste0("../testdata/", file)
+  app$setInputs(SCD_es_calculator = "Batch Entry")
+
+  if (str_detect(file, "csv")) {
+    
+    app$setInputs(dat_type = "dat")
+    app$uploadFile(dat = data_path)
+    
+  } else if (str_detect(file, "xlsx")) {
+    
+    app$setInputs(dat_type = "xlsx")
+    app$uploadFile(xlsx = data_path)
+    
+  }
+  
+  app$setInputs(BatchEntryTabs = "Variables")
+  app$setInputs(b_clusters = "Case_pseudonym")
+  app$setInputs(b_phase = "Condition")
+  app$setInputs(session_number = "Session_number")
+  app$setInputs(b_out = "Outcome")
+  app$setInputs(bimprovement = "decrease")
+  app$setInputs(BatchEntryTabs = "Estimate")
+  app$setInputs(bESno = c("IRD", "NAP", "PAND", "PEM", "PND", "Tau", "Tau_BC", "Tau_U"))
+  app$setInputs(bESpar = c("LOR", "LRRd", "LRRi", "LRM", "SMD"))
+  app$setInputs(boutScale = "count")
+  app$setInputs(batchest = "click")
+  
+  output_app <- app$getValue(name = "batchTable")
+  output_app_table <-
+    as.data.frame(rvest::read_html(output_app) %>% rvest::html_table(fill = TRUE)) %>%
+    mutate(across(Est:CI_upper, ~ ifelse(. == "-", NA, .))) %>%
+    mutate(across(Est:CI_upper, as.numeric))
+  
+  return(output_app_table)
+
+}
+
+
+test_that("Data are uploaded correctly.", {
+  
+  skip_on_cran()
+  
+  # csv file
+  output_csv <- check_load("McKissick.csv")
+  
+  # excel file
+  output_xlsx <- check_load("McKissick.xlsx")
```

Comment:
I used McKissick data for checking both excel and csv file uploading. Thought this makes the code simpler compared to using two different data sets for excel and csv uploading checking.